### ISSUE TITLE
error: static assertion failed: tvin may only be used with operators =, == and !=

### ISSUE THREAD

**Issue by markwright** [rbock/sqlpp11#31]
Hi, sqlpp11 is really neat, I encounter  compiler error with the following test case with 0.31,
while as I think this will compile and run with 0.29.  Thanks, Mark

tvin.cpp:

````
#include <sqlpp11/sqlpp11.h>
#include <sqlpp11/sqlite3/sqlite3.h>
#include "tab.h"

using sqlpp_connection = sqlpp::sqlite3::connection;
using sqlpp_connection_config = sqlpp::sqlite3::connection_config;

int main(int argc, char *argv[])
{
	sqlpp_connection_config config;
 	config.path_to_database = ":memory:";
	config.flags = SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
	config.debug = true;
	sqlpp_connection db(config);
  db.execute(R"(CREATE TABLE tab (
    id integer not null,
    str character varying(10)
    ))");
  T::Tab t;
  db(remove_from(t).where(true));
  std::string s;
  db(sqlpp::insert_into(t).set(t.id = 1, t.str = sqlpp::tvin(s)));
  return 0;
}
````

tab.sql:

````
CREATE TABLE tab (
  id integer not null,
  str varchar
);
````

Makefile:

````
tvin:	tvin.cpp tab.h
	g++ -std=gnu++14 tvin.cpp

tab.h:	tab.sql
	ddl2cpp tab.sql tab T
````

Compiler error with g++-4.9.2:

````
-*- mode: compilation; default-directory: "/data/eng/dev/cxx/sqlpp/tvin/" -*-
Compilation started at Fri Mar 13 21:33:55

make
g++ -std=gnu++14 tvin.cpp
In file included from /usr/include/sqlpp11/boolean.h:35:0,
                 from /usr/include/sqlpp11/column_types.h:30,
                 from /usr/include/sqlpp11/sqlpp11.h:31,
                 from tvin.cpp:1:
/usr/include/sqlpp11/tvin.h: In instantiation of 'static void sqlpp::assert_tvin_with_correct_operator_t::_() [with T = void]':
/usr/include/sqlpp11/sqlite3/connection.h:282:60:   required from 'decltype (t._run((*(sqlpp::sqlite3::connection*)this))) sqlpp::sqlite3::connection::operator()(const T&) [with T = sqlpp::statement_t<void, sqlpp::insert_t, sqlpp::into_t<void, T::Tab>, sqlpp::insert_list_t<void, sqlpp::assignment_t<sqlpp::column_t<T::Tab, T::Tab_::Id>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<T::Tab, T::Tab_::Str>, sqlpp::tvin_arg_t<sqlpp::text_operand> > > >; decltype (t._run((*(sqlpp::sqlite3::connection*)this))) = long unsigned int]'
tvin.cpp:22:65:   required from here
/usr/include/sqlpp11/tvin.h:66:4: error: static assertion failed: tvin may only be used with operators =, == and !=
    static_assert(wrong_t<T>::value, "tvin may only be used with operators =, == and !=");
    ^
Makefile:2: recipe for target 'tvin' failed
make: *** [tvin] Error 1

Compilation exited abnormally with code 2 at Fri Mar 13 21:33:56
````

**Comment by rbock** [comment]
I think that should work. I'll see what I can do.

**Comment by rbock** [comment]
Fixed in 0.32. Thanks for the report!
